sunpy/net/dataretriever/sources/xrt.py

```diff
@@ -0,0 +1,111 @@
+    Returns a list of URLS to XRT files corresponding to value of input timerange.
+    URL source: `http://solar.physics.montana.edu/HINODE/XRT/QL/syn_comp_fits/`.
+    Parameters
+    ----------
+    timerange: sunpy.time.TimeRange
+        time range for which data is to be downloaded.
+        Example value - TimeRange('2015-12-30 00:00:00','2015-12-31 00:01:00')
+    
+    Instrument: Fixed argument = 'xrt'
+    Filter: Filter can take only 'almesh', 'alpoly', 'cpoly', 'tipoly', 'thinbe' as arguments.
+            Arguments are case-insensitive and can include non alphabetic characters as well.
+            Arguments like 'Al-mesh' or 'Al_mesh' or 'Al/mesh' would also work fine.
+            
+    Returns
+    -------
+    urls: list
+    list of urls corresponding to requested time range.
+    
+    Examples
+    --------
+    >>> from sunpy.net import Fido
+    >>> from sunpy.net import attrs as a
+    >>> results = Fido.search(Time('2016/5/18','2016/5/19'), a.Instrument('xrt'), a.Filter('al_mesh'))
+    >>> print(results)
+        [<Table length=2>
+         Start Time           End Time      Source Instrument
+           str19               str19         str6     str3   
+    ------------------- ------------------- ------ ----------
+    2016-05-18 00:00:00 2016-05-19 00:00:00 Hinode        XRT
+    2016-05-19 00:00:00 2016-05-20 00:00:00 Hinode        XRT]
+    
+    >>> response = Fido.fetch(results)
+    """
+    def _get_url_for_timerange(self, timerange, **kwargs):
+        """
+        returns list of urls corresponding to given TimeRange.
+        """
+        if not timerange:
+            return []      
+        START_DATE = datetime.datetime(2014, 1, 10, 18, 14, 42)
+        filter_type = kwargs['filter']
+        regex = re.compile('[^a-zA-Z]')
+        filter_type = regex.sub('', filter_type).lower()
+        result = list()
+        if timerange.start < START_DATE:
+            raise ValueError('Earliest date for which XRT data is available is {:%Y-%m-%d}'.format(START_DATE))
+        filter_dict = {'almesh':'Al_mesh', 'alpoly':'Al_poly', 'cpoly':'C_poly', 'tipoly':'Ti_poly', 'thinbe':'thin_Be'}
+        url_pattern = ['http://solar.physics.montana.edu/HINODE/XRT/QL/syn_comp_fits/XRT_{filter}_%Y%m%d_%H%M%S.{s}.fits'.
+                       format(filter = filter_dict[filter_type], s=i) for i in range (0, 10)]
+        arr = [Scraper(pattern, filter = filter_dict[filter_type]).filelist(timerange) for pattern in url_pattern]
+        [result.extend(url) for url in arr if len(url)>0]
+        #Integrate nascom.nasa.gov XRT data as well.
+        url_pattern = ['http://sohowww.nascom.nasa.gov/sdb/hinode/xrt/l1q_synop/XRT_{filter}_%Y%m%d_%H%M%S.{s}.fits'.
+                       format(filter = filter_dict[filter_type], s=i) for i in range (0, 10)]
+        arr = [Scraper(pattern, filter = filter_dict[filter_type]).filelist(timerange) for pattern in url_pattern]
+        [result.extend(url) for url in arr if len(url)>0]
```

Comment:
[I have replied here that it would be incorrect and the tests fail when I do that](https://github.com/sunpy/sunpy/pull/1827#discussion_r67875085). We are having a small communication problem, whenever you ask me a question about something on GitHub regarding a client and I reply to that and commit something new, my reply is getting lost in all these comments on ```outdated diffs```. And since the same thing is done across multiple clients, you would assume I haven't corrected them in all when I have. Some of the questions you have asked I've already replied to but I have to post the replies once again (like in this case).